app/actions/player.js

```diff
@@ -58,6 +61,31 @@ export function updateVolume(volume) {
   };
 }
 
+export function muteOn() {
+  return {
+    type: MUTE_ON,
+    payload: null
+  };
+}
+
+export function muteOff() {
+  return {
+    type: MUTE_OFF,
+    payload: null
+  };
+}
+
+export function toggleMute(muted) {
+  return dispatch => {
+    if (muted) {
+      dispatch(muteOn());
+      dispatch(updateVolume(0));
```

Comment:
I think this line can (and should be removed), we won't be able to restore previous volume on unmuting otherwise.

Reply:
@nukeop Removing the `dispatch(updateVolume(0))`  wont work. Initially when I started implementing the ability to mute volume when clicking on the volume icon, I looked at how the existing sound output works. I came across `SoundContainer` which renders a `Sound` component from `react-sound` npm package which ultimately uses the `SoundManager2` library. The `react-sound` package does not accept a flag to `mute` or `unmute` the volume. The way to implemet currently is by passing in 0 for volume. This is the reason I introduced state for the `VolumeControl` component to save the volume slider's value.

Reply:
Check my comment in your pull request, we'll need to edit another line to make it work.

Reply:
Yep missed the bit there. Thanks for that. Will push the changes soon.

Reply:
Also, do we want to change from `mute` to `unmute` if the user changes the volume slider?